go-controller/pkg/cni/helper_linux.go

```diff
@@ -252,60 +252,81 @@ func setupInterface(netns ns.NetNS, containerID, ifName string, ifInfo *PodInter
 	return hostIface, contIface, nil
 }
 
+// a unique ifname is generated from the containerID by adding a variable suffix
+// for convention:
+// port representors: suffixes are _r1 (primary iface) and _r2 (secondary iface)
+// interfaces to be moved to the pod: suffixes are _1 (primary iface) and _2 (secondary iface)
+func generateIfName(containerID string, isRepresentor bool, isPrimary bool) string {
+	ifnameSuffix := "_"
+	if isRepresentor {
+		ifnameSuffix += "r"
+	}
+	if isPrimary {
+		ifnameSuffix += "1"
+	} else {
+		ifnameSuffix += "2"
```

Comment:
this might be a dumb question, but what if a pod has multiple secondary attachments to  sriov devices ?... 

Shouldn't we pass the iface index or something like that to ensure that scenario works ?

Reply:
I would prefer to have two functions: `generateIfName` + `generateRepresentorName`